simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -36,96 +37,119 @@
      */
     @Inject
     public BlockStreamSimulatorApp(
             @NonNull Configuration configuration,
             @NonNull BlockStreamManager blockStreamManager,
             @NonNull PublishStreamGrpcClient publishStreamGrpcClient,
+            @NonNull ConsumerStreamGrpcClient consumerStreamGrpcClient,
             @NonNull MetricsService metricsService) {
 
         requireNonNull(configuration);
         requireNonNull(blockStreamManager);
+
         this.metricsService = requireNonNull(metricsService);
         this.publishStreamGrpcClient = requireNonNull(publishStreamGrpcClient);
+        this.consumerStreamGrpcClient = requireNonNull(consumerStreamGrpcClient);
+        this.isRunning = new AtomicBoolean(false);
+
+        // Initialize the appropriate mode handler based on configuration
         final BlockStreamConfig blockStreamConfig =
                 requireNonNull(configuration.getConfigData(BlockStreamConfig.class));
-
         final SimulatorMode simulatorMode = blockStreamConfig.simulatorMode();
-        switch (simulatorMode) {
-            case PUBLISHER -> simulatorModeHandler = new PublisherModeHandler(
+        this.simulatorModeHandler = switch (simulatorMode) {
+            case PUBLISHER -> new PublisherModeHandler(
                     blockStreamConfig, publishStreamGrpcClient, blockStreamManager, metricsService);
-            case CONSUMER -> simulatorModeHandler = new ConsumerModeHandler(blockStreamConfig);
-            case BOTH -> simulatorModeHandler = new CombinedModeHandler(blockStreamConfig);
+            case CONSUMER -> new ConsumerModeHandler(blockStreamConfig, consumerStreamGrpcClient);
+            case BOTH -> new CombinedModeHandler(blockStreamConfig);
             default -> throw new IllegalArgumentException("Unknown SimulatorMode: " + simulatorMode);
-        }
+        };
     }
 
     /**
-     * Starts the block stream simulator.
+     * Initializes and starts the block stream simulator in the configured mode.
+     * This method initializes all components and begins the streaming process.
      *
-     * @throws InterruptedException if the thread is interrupted
-     * @throws BlockSimulatorParsingException if a parse error occurs
-     * @throws IOException if an I/O error occurs
+     * @throws InterruptedException if the streaming process is interrupted
+     * @throws BlockSimulatorParsingException if a block parsing error occurs
+     * @throws IOException if an I/O error occurs during streaming
      */
     public void start() throws InterruptedException, BlockSimulatorParsingException, IOException {
-        LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator started initializing components...");
-        publishStreamGrpcClient.init();
+        LOGGER.log(INFO, "Block Stream Simulator started initializing components...");
+        simulatorModeHandler.init();
+
         isRunning.set(true);
 
         simulatorModeHandler.start();
```

Comment:
Shouldn't the `isRunning` field be set to `true` after the starting of the `simulatorModeHandler`?

Reply:
No because start is a synchronous process, which will end when esentially the simulator is no longer running, therefore we set it to true just before we enter.